StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1204StaticElementsMustAppearBeforeInstanceElements.cs

```diff
@@ -79,70 +62,126 @@ private static void HandleCompilationStart(CompilationStartAnalysisContext conte
             context.RegisterSyntaxNodeActionHonorExclusions(TypeDeclarationAction, TypeDeclarationKinds);
         }
 
-        private static void HandleCompilationUnit(SyntaxNodeAnalysisContext context)
+        private static void HandleCompilationUnit(SyntaxNodeAnalysisContext context, StyleCopSettings settings)
         {
+            var elementOrder = settings.OrderingRules.ElementOrder;
+            int staticIndex = elementOrder.IndexOf(OrderingTrait.Static);
+            if (staticIndex < 0)
+            {
+                return;
+            }
+
             var compilationUnit = (CompilationUnitSyntax)context.Node;
 
-            HandleMemberList(context, compilationUnit.Members, AccessLevel.Internal);
+            HandleMemberList(context, elementOrder, staticIndex, compilationUnit.Members, AccessLevel.Internal);
         }
 
-        private static void HandleNamespaceDeclaration(SyntaxNodeAnalysisContext context)
+        private static void HandleNamespaceDeclaration(SyntaxNodeAnalysisContext context, StyleCopSettings settings)
         {
+            var elementOrder = settings.OrderingRules.ElementOrder;
+            int staticIndex = elementOrder.IndexOf(OrderingTrait.Static);
+            if (staticIndex < 0)
+            {
+                return;
+            }
+
             var compilationUnit = (NamespaceDeclarationSyntax)context.Node;
 
-            HandleMemberList(context, compilationUnit.Members, AccessLevel.Internal);
+            HandleMemberList(context, elementOrder, staticIndex, compilationUnit.Members, AccessLevel.Internal);
         }
 
-        private static void HandleTypeDeclaration(SyntaxNodeAnalysisContext context)
+        private static void HandleTypeDeclaration(SyntaxNodeAnalysisContext context, StyleCopSettings settings)
         {
+            var elementOrder = settings.OrderingRules.ElementOrder;
+            int staticIndex = elementOrder.IndexOf(OrderingTrait.Static);
+            if (staticIndex < 0)
+            {
+                return;
+            }
+
             var typeDeclaration = (TypeDeclarationSyntax)context.Node;
 
-            HandleMemberList(context, typeDeclaration.Members, AccessLevel.Private);
+            HandleMemberList(context, elementOrder, staticIndex, typeDeclaration.Members, AccessLevel.Private);
         }
 
-        private static void HandleMemberList(SyntaxNodeAnalysisContext context, SyntaxList<MemberDeclarationSyntax> members, AccessLevel defaultAccessLevel)
+        private static void HandleMemberList(SyntaxNodeAnalysisContext context, ImmutableArray<OrderingTrait> elementOrder, int staticIndex, SyntaxList<MemberDeclarationSyntax> members, AccessLevel defaultAccessLevel)
```

Comment:
:thought_balloon: The `defaultAccessLevel` parameter can be removed, as it is not used anymore.